hbase-server/src/main/java/org/apache/hadoop/hbase/master/cleaner/DirScanPool.java

```diff
@@ -39,21 +39,28 @@
   private final ThreadPoolExecutor pool;
   private int cleanerLatch;
   private boolean reconfigNotification;
+  private final String cleanerPoolConfig;
+  private final String cleanerPoolConfigDefault;
+  private final String name;
 
-  public DirScanPool(Configuration conf) {
-    String poolSize = conf.get(CleanerChore.CHORE_POOL_SIZE, CleanerChore.DEFAULT_CHORE_POOL_SIZE);
+  private DirScanPool(Configuration conf, String cleanerPoolConfig, String cleanerPoolConfigDefault,
```

Comment:
CleanerPoolConfig is too general and I thought it was a class...

Let's just name them cleanerPoolSizeConfigName, and cleanerPoolSizeConfigDefault?

Reply:
+1 for this suggested name. Looks clean.
Can we think of making this more clean?
DirScanPool can have a private Enum Type (for ArchivedHFilesCleaner, OldLogsCleaner)
The constructor can only take the type.  So the other details of name, configName, configDefault are considered based on type.  Infact type can have a method to return the poolSize.
We dont have to even store the cleanerPoolConfigDefault
I think its used only on configUpdate method.   That can use Configuration#get(String) and have a null check?

Reply:
+1 for "DirScanPool can have a private Enum Type (for ArchivedHFilesCleaner, OldLogsCleaner)"